tests/checkpointing/test_checkpoint_conversion.py

```diff
@@ -0,0 +1,101 @@
+import os
+from pathlib import Path
```

Comment:
type hints are missing basically everywhere in the tests. That makes it difficult to verify the correctness of the tests.